uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsencePeriod.php

```diff
@@ -581,4 +581,41 @@ public static function getPeriodContainingDates(DateTime $fromDate, DateTime $to
     }
     return null;
   }
+
+  /**
+   * Returns the absence periods overlapping the given start and
+   * end dates ordered by the weight in ascending order. When
+   * the end date is not present, it returns only the absence
+   * periods with end dates >= to the start date.
```

Comment:
Agreed, Although the method does what its supposed to do as required by the Public Holiday service, maybe the name is wrong? or am I lumping two functionalities in one function?

Reply:
I was looking at the code again and you're using this method in only two places:

`PublicHolidayLeaveRequestCreation.createAllForContract`
`PublicHolidayLeaveRequestCreation.createForAllInTheFuture`

In the former, you actually need periods overlapping dates, but for the latter, you need something slightly different. What you need there is more of a list of absence periods after or at a specific date. 

I know the code/conditions for both are very similar, but I think it would be good to have separate methods for them to make this criteria clear and avoid confusions